### PR TITLE
Add the water bucket clutch type

### PR DESCRIPTION
**Pull request by ellieisjelly**
Adds a clutch type using the water bucket.
Kind of janky but it works

### CODE REVIEW

File: src/main/java/io/github/haykam821/clutchpractice/clutch/PlacementClutchType.java [hunk 1]
```diff
@@ -7,28 +7,40 @@
 import io.github.haykam821.clutchpractice.game.map.ClutchPracticeMap;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
+import net.minecraft.fluid.Fluid;
 import net.minecraft.item.ItemStack;
 import net.minecraft.server.world.ServerWorld;
 import net.minecraft.text.Text;
 import xyz.nucleoid.plasmid.util.ItemStackBuilder;
 
 public class PlacementClutchType extends ClutchType {
```
Comment (haykam821) [inline src/main/java/io/github/haykam821/clutchpractice/clutch/PlacementClutchType.java:15]: Would it be possible to instead split this class into a `BlockPlacementClutchType` based on the block name and item and `FluidPlacementClutchType` based on the bucket name and item? Alternatively, I believe blocks and fluids can be ignored in place of only using item stacks.

File: src/main/java/io/github/haykam821/clutchpractice/clutch/ClutchType.java [hunk 2]
```diff
@@ -16,7 +16,6 @@
 public abstract class ClutchType {
 	private final ItemStack icon;
 	private Text name;
-
```
Comment (haykam821) [inline src/main/java/io/github/haykam821/clutchpractice/clutch/ClutchType.java]: Make sure to keep spacing between methods and fields.

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/haykam821/clutchpractice/clutch/PlacementClutchType.java (hunk 1 above)
Comment (ellieisjelly) [reply src/main/java/io/github/haykam821/clutchpractice/clutch/PlacementClutchType.java:15]: Going to go with the latter, as I don't see any reason why we need the separation when ItemStacks can accomplish the same while being more universal

Comment (ellieisjelly) [reply src/main/java/io/github/haykam821/clutchpractice/clutch/PlacementClutchType.java:15]: New commit fixes this

File: src/main/java/io/github/haykam821/clutchpractice/clutch/ClutchType.java (hunk 2 above)
Comment (ellieisjelly) [reply src/main/java/io/github/haykam821/clutchpractice/clutch/ClutchType.java]: New commit fixes this

**Review by haykam821 — APPROVED** [review]
Thank you for the pull request!

**Pull request merged**